Frends.Kafka.Produce/Frends.Kafka.Produce/Produce.cs

```diff
@@ -0,0 +1,196 @@
+            config.SaslOauthbearerTokenEndpointUrl = String.IsNullOrWhiteSpace(sasl.SaslOauthbearerTokenEndpointUrl) ? "" : sasl.SaslOauthbearerTokenEndpointUrl;
+            config.SaslOauthbearerConfig = String.IsNullOrWhiteSpace(sasl.SaslOauthbearerConfig) ? "" : sasl.SaslOauthbearerConfig;
+            config.SaslOauthbearerExtensions = String.IsNullOrWhiteSpace(sasl.SaslOauthbearerExtensions) ? "" : sasl.SaslOauthbearerExtensions;
+            config.SaslOauthbearerScope = String.IsNullOrWhiteSpace(sasl.SaslOauthbearerScope) ? "" : sasl.SaslOauthbearerScope;
+            config.SaslKerberosKeytab = String.IsNullOrWhiteSpace(sasl.SaslKerberosKeytab) ? "" : sasl.SaslKerberosKeytab;
+            config.SaslKerberosMinTimeBeforeRelogin = sasl.SaslKerberosMinTimeBeforeRelogin;
+            config.SaslKerberosPrincipal = sasl.SaslKerberosPrincipal;
+            config.SaslKerberosServiceName = sasl.SaslKerberosServiceName;
+        }
+
+        if (ssl.UseSSL)
+        {
+            config.SslEndpointIdentificationAlgorithm = SetSslEndpointIdentificationAlgorithm(ssl);
+            config.EnableSslCertificateVerification = ssl.EnableSslCertificateVerification;
+            config.SslCertificateLocation = String.IsNullOrWhiteSpace(ssl.SslCertificateLocation) ? "" : ssl.SslCertificateLocation;
+            config.SslCertificatePem = String.IsNullOrWhiteSpace(ssl.SslCertificatePem) ? "" : ssl.SslCertificatePem;
+            config.SslCaCertificateStores = ssl.SslCaCertificateStores;
+            config.SslCaLocation = String.IsNullOrWhiteSpace(ssl.SslCaLocation) ? "" : ssl.SslCaLocation;
+            config.SslCaPem = String.IsNullOrWhiteSpace(ssl.SslCaPem) ? "" : ssl.SslCaPem;
+            config.SslKeyLocation = String.IsNullOrWhiteSpace(ssl.SslKeyLocation) ? "" : ssl.SslKeyLocation;
+            config.SslKeyPassword = String.IsNullOrWhiteSpace(ssl.SslKeyPassword) ? "" : ssl.SslKeyPassword;
+            config.SslKeyPem = String.IsNullOrWhiteSpace(ssl.SslKeyPem) ? "" : ssl.SslKeyPem;
+            config.SslKeystoreLocation = String.IsNullOrWhiteSpace(ssl.SslKeystoreLocation) ? "" : ssl.SslKeystoreLocation;
+            config.SslKeystorePassword = String.IsNullOrWhiteSpace(ssl.SslKeystorePassword) ? "" : ssl.SslKeystorePassword;
+            config.SslEngineLocation = String.IsNullOrWhiteSpace(ssl.SslEngineLocation) ? "" : ssl.SslEngineLocation;
+            config.SslCipherSuites = String.IsNullOrWhiteSpace(ssl.SslCipherSuites) ? "" : ssl.SslCipherSuites;
+            config.SslCrlLocation = String.IsNullOrWhiteSpace(ssl.SslCrlLocation) ? "" : ssl.SslCrlLocation;
+            config.SslCurvesList = String.IsNullOrWhiteSpace(ssl.SslCurvesList) ? "" : ssl.SslCurvesList;
+            config.SslSigalgsList = String.IsNullOrWhiteSpace(ssl.SslSigalgsList) ? "" : ssl.SslSigalgsList;
+        }
+
+        return config;
+    }
+    private static SecurityProtocol SetSecurityProtocol(Input input)
```

Comment:
The usual convention is that any method starting with "Set" should actually set something. In this method (and all methods below) you do not set anything in the method but actually just map values. So a "Get" would be more appropriate.